common/network_manager_utils.go

```diff
@@ -0,0 +1,76 @@
+package common
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	k6lib "go.k6.io/k6/lib"
+	k6netext "go.k6.io/k6/lib/netext"
+	k6types "go.k6.io/k6/lib/types"
+)
+
+func checkBlockedHosts(host string, blockedHosts *k6types.HostnameTrie) error {
+	if blockedHosts == nil {
+		return nil
+	}
+	if match, blocked := blockedHosts.Contains(host); blocked {
+		return fmt.Errorf("hostname %s is in a blocked pattern (%s)", host, match)
+	}
+	return nil
+}
+
+func checkBlockedIPs(ip net.IP, blockedIPs []*k6lib.IPNet) error {
+	for _, ipnet := range blockedIPs {
+		if ipnet.Contains(ip) {
+			// TODO: Return netext.BlackListedIPError here once its private
+			// fields are exported, or there's a constructor for it.
+			return fmt.Errorf("IP (%s) is in a blacklisted range (%s)", ip, ipnet)
```

Comment:
```suggestion
			return fmt.Errorf("IP (%q) is in a blacklisted range (%q)", ip, ipnet)
```

Reply:
Similar to above, we don't need both parenthesis and quotes in this case. I would actually drop the parenthesis in the first case and change it to `IP %s is in a blacklisted range (%s)`, but note that I left the message the same as the one in `BlackListedIPError`, so maybe it's best if we leave this as is as well.

Reply:
Fair enough, here you go:

```suggestion
			return fmt.Errorf("IP %s is in a blacklisted range %q", ip, ipnet)
```

Reply:
Same comment as above, but changed in b8eef0e.